vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import com.airbnb.mvrx.withState
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.tabs.TabLayoutMediator
+import im.vector.app.BuildConfig
+import im.vector.app.R
+import im.vector.app.databinding.FragmentFtueSplashCarouselBinding
+import im.vector.app.features.VectorFeatures
+import im.vector.app.features.onboarding.OnboardingAction
+import im.vector.app.features.onboarding.OnboardingFlow
+import im.vector.app.features.settings.VectorPreferences
+import org.matrix.android.sdk.api.failure.Failure
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class FtueAuthSplashCarouselFragment @Inject constructor(
+        private val vectorPreferences: VectorPreferences,
+        private val vectorFeatures: VectorFeatures,
+        private val carouselController: SplashCarouselController
+) : AbstractFtueAuthFragment<FragmentFtueSplashCarouselBinding>() {
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueSplashCarouselBinding {
+        return FragmentFtueSplashCarouselBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    private fun setupViews() {
+        views.splashCarousel.adapter = carouselController.adapter
+        TabLayoutMediator(views.carouselIndicator, views.splashCarousel) { _, _ -> }.attach()
+        carouselController.setData(SplashCarouselState(
+                items = listOf(
```

Comment:
Since this state is static, maybe move all this to the constructor of `SplashCarouselState`?

Reply:
https://github.com/vector-im/element-android/pull/4727/commits/8a3652aec2cf9107a22336074be6388d9380e8e9